### PR TITLE
gdal 1.11.2_2 java swig bindings #39105

### PR DESCRIPTION
**Pull request by pauldaustin**
Added the --with-swig-java option to enable compilation of the swig
java bindings for gdal. Added the code to create the required
java config file for java 1.7 and run make on swig java binding.

### CODE REVIEW

**Comment by pauldaustin** [comment]
Is there anything I need to configure on my clone to allow the merge to work?

File: Library/Formula/gdal.rb [hunk 1]
```diff
@@ -70,6 +71,8 @@ class Gdal < Formula
     depends_on "json-c"
   end
 
+  depends_on 'swig'
```
Comment (mikemcquaid) [inline Library/Formula/gdal.rb]: Probably needs a `if build.with? "swig"` on the end?

File: Library/Formula/gdal.rb [hunk 2]
```diff
@@ -284,6 +287,19 @@ def install
...
     end
 
+    if 'with-swig-java'
+      cd 'swig/java' do
+        system "echo 'JAVA_HOME='`/usr/libexec/java_home -v 1.7` > java.opt"
+        system "echo 'JAVADOC=$(JAVA_HOME)/bin/javadoc' >> java.opt"
+        system "echo 'JAVAC=$(JAVA_HOME)/bin/javac' >> java.opt"
+        system "echo 'JAVA=$(JAVA_HOME)/bin/java' >> java.opt"
+        system "echo 'JAR=$(JAVA_HOME)/bin/jar' >> java.opt"
+        system "echo 'JAVA_INCLUDE=-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/darwin' >> java.opt"
```
Comment (mikemcquaid) [inline Library/Formula/gdal.rb]: Use `Pathname.write` instead of `echo`.

File: Library/Formula/gdal.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ class Gdal < Formula
   homepage 'http://www.gdal.org/'
   url "http://download.osgeo.org/gdal/1.11.2/gdal-1.11.2.tar.gz"
   sha1 "6f3ccbe5643805784812072a33c25be0bbff00db"
-  revision 1
+  revision 2
```
Comment (mikemcquaid) [inline Library/Formula/gdal.rb]: We don't need this as we're not changing the default options.

**Comment by pauldaustin** [comment]
The gdal swig java makefile requires a JAVA_HOME to be available in the makefile. I can pass this into the make command using make JAVA_HOME=$JAVA_HOME rather than editing the java.opt.

Unfortunately the brew audit command looks for JAVA_HOME and fails it.

How should I be passing in the JAVA_HOME to the makefile.

**Comment by pauldaustin** [comment]
Just curious once the merged build finishes successfully what's the process for getting that committed to the master branch?

File: Library/Formula/gdal.rb [hunk 4]
```diff
@@ -284,6 +290,15 @@ def install
       bin.install Dir['scripts/*']
     end
 
+    if build.with? "swig-java"
+      cd 'swig/java' do
+        inreplace "java.opt", "linux", "darwin"
```
Comment (mikemcquaid) [inline Library/Formula/gdal.rb]: Does this official support OS X?

File: Library/Formula/gdal.rb [hunk 5]
```diff
@@ -308,3 +323,4 @@ def caveats
     system "#{bin}/ogrinfo", "--formats"
   end
 end
+
```
Comment (mikemcquaid) [inline Library/Formula/gdal.rb]: Doesn't need this newline.

File: Library/Formula/gdal.rb [hunk 6]
```diff
@@ -70,6 +71,11 @@ class Gdal < Formula
     depends_on "json-c"
   end
 
+  depends_on :java => ["1.7+", :optional, :build]
+  if build.with? "swig-java"
+    depends_on 'swig'
```
Comment (mikemcquaid) [inline Library/Formula/gdal.rb]: `depends_on "swig" if build.with? "swig-java"`

File: Library/Formula/gdal.rb (hunk 4 above)
Comment (pauldaustin) [inline Library/Formula/gdal.rb]: The default java.opt (config file) is for linux but the build can support macosx

The in replace changes this

    JAVA_INCLUDE=-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux

to this

    JAVA_INCLUDE=-I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/darwin

File: Library/Formula/gdal.rb (hunk 5 above)
Comment (pauldaustin) [inline Library/Formula/gdal.rb]: In the Ruby Style guide https://github.com/styleguide/ruby referenced from https://github.com/Homebrew/homebrew/tree/master/share/doc/homebrew#readme

End each file with a blank newline.

Comment (mikemcquaid) [inline Library/Formula/gdal.rb]: There already is one in this file, the GitHub UI and most text editors don't show it.

Comment (pauldaustin) [inline Library/Formula/gdal.rb]: I've never noticed that about vi hiding that extra newline before. Opened it up in TextWranger with invisibles shown and yup there was two line feeds. Fixed it now.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `gdal: add java swig bindings.`.

**Pull request closed without merging**